kite-data/kite-data-core/src/main/java/org/kitesdk/data/ReadiableView.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2015 Cloudera.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kitesdk.data;
+
+/**
+ * Readiable views can be used to signal to consumers that a particular view is
```

Comment:
Maybe this is a bit of a nitpick, but the description seems a bit awkward to me.

"Readiable views can be used to signal..."

It almost sounds like if you want to signal that some arbitrary view X is ready then you need to find some other (readiable) view Y whose purpose is to give you the ability to signal that X is ready.

The purpose of a ReadiableView isn't to signal.  The purpose is to provide access to a subset of some Dataset.

I would suggest something like "A Readiable view is a view that may signal consumers when the underlying data is ready for consumption.  Not all View implementations provide this capability."

Furthermore, although the interface is named ReadiableView, there are no methods on the interface that actually represent a View.  As such, something could implement this interface and not be a View at all.  With the given definition I think the interface would more appropriately be named simply "Readiable".  Alternatively, the interface could extend View such that it really does represent something that is both a View and Readiable.